components/nimbus/examples/experiment.rs

```diff
@@ -218,7 +218,10 @@ fn main() -> Result<()> {
 
     // initiate the optional config
     let config = RemoteSettingsConfig {
-        server_url: Some(server_url.to_string()),
+        server: Some(RemoteSettingsServer::Custom {
+            url: server_url.to_string(),
+        }),
+        server_url: None,
         bucket_name: None,
         collection_name: collection_name.to_string(),
```

Comment:
What do you think about using `..RemoteSettingsConfig::default()` at the end of the struct initialization?  It could be a nice way to handle `server_url` being deprecated and eventually being removed.

Reply:
Hmmm, I don't think that'll work, because `RemoteSettingsConfig` doesn't implement `Default`. We could have it derive `Default`, but we don't have a suitable default that we can use for `collection_name`, since that's the only setting that _is_ required.

I wonder if moving `RemoteSettingsConfig` to a builder-style API would be a good long-term direction, though?

```rs
let b = RemoteSettingsConfig::with_collection_name("quicksuggest".into());
b.server(RemoteSettingsServer::Stage)?.bucket_name("main".into()); // Optional.
let client = b.build();
```

Reply:
We could even do that in a backward-compatible way by renaming `RemoteSettingsConfig` to `RemoteSettingsBuilder`, deprecating `RemoteSettingsConfig`, and migrating callers over to `RemoteSettingsBuilder`.

I'm really liking the builder pattern you added for Suggest!